File: Programmers/level2/camouflage/camouflage.py
# 문제 출처 : https://programmers.co.kr/learn/courses/30/lessons/42578?language=python3

from functools import reduce


def solution(clothes):
    obj = {}
    for i in range(len(clothes)):
        if clothes[i][1] in obj:
            obj[clothes[i][1]] += 1
        else:
            obj[clothes[i][1]] = 1
    print(obj)
    answer = reduce(lambda x, y: x*(y+1), obj.values(), 1) - 1
    return answer

File: Programmers/level2/camouflage/camouflage.js
// https://programmers.co.kr/learn/courses/30/lessons/42578?language=javascript

export const solution = (clothes) => {
    // var answer = 1;
    // let obj = {};
    // for(let i = 0; i < clothes.length; i++){
    //     obj[clothes[i][1]] = (obj[clothes[i][1]] || 1) + 1
    // }
    // console.log(obj)
    // for(let z in obj){
    //     answer *= obj[z]
    // }
    // return answer - 1;
    return Object.values(clothes.reduce((obj,z) => {
        obj[z[1]] = obj[z[1]] ? obj[z[1]] + 1 : 1;
        console.log(obj)
        return obj
    }, {})).reduce((a,b) => a*(b+1),1)-1;
}